driver/controller.go

```diff
@@ -181,20 +181,13 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 		volumeReq.SnapshotID = snapshotID
 	}
 
-	log.Info("checking volume limit")
-	details, err := d.checkLimit(ctx)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to check volume limit: %s", err)
-	}
-	if details != nil {
-		return nil, status.Errorf(codes.ResourceExhausted,
-			"volume limit (%d) has been reached. Current number of volumes: %d. Please contact support.",
-			details.limit, details.numVolumes)
-	}
-
 	log.WithField("volume_req", volumeReq).Info("creating volume")
-	vol, _, err := d.storage.CreateVolume(ctx, volumeReq)
+	vol, cvResp, err := d.storage.CreateVolume(ctx, volumeReq)
 	if err != nil {
+		if cvResp != nil && cvResp.Response != nil && cvResp.StatusCode == http.StatusForbidden && strings.Contains(err.Error(), "capacity limit exceeded") {
```

Comment:
Small nit / simplification suggestion: I think it's common to check the outer `*godo.Response` for nil-ness only, not the inner `*http.Response`  [since godo always populates them together](https://github.com/digitalocean/godo/blob/2e7120f15645a6399cb37a949bf9a6168b4b6212/godo.go#L378).

Reply:
Sure, I've removed the `cvResp.Response != nil` check and only kept `cvResp != nil`.